java-spiffe-core/src/main/java/io/spiffe/svid/x509svid/X509Svid.java

```diff
@@ -0,0 +1,169 @@
+package io.spiffe.svid.x509svid;
+
+import io.spiffe.exception.X509SvidException;
+import io.spiffe.internal.CertificateUtils;
+import io.spiffe.spiffeid.SpiffeId;
+import lombok.NonNull;
+import lombok.Value;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateParsingException;
+import java.security.cert.X509Certificate;
+import java.security.spec.InvalidKeySpecException;
+import java.util.List;
+
+/**
+ * Represents a SPIFFE X.509 SVID.
+ * <p>
+ * Contains a SPIFFE ID, a private key and a chain of X.509 certificates.
+ */
+@Value
+public class X509Svid implements X509SvidSource {
+
+    SpiffeId spiffeId;
+
+    /**
+     * The X.509 certificates of the X.509-SVID. The leaf certificate is
+     * the X.509-SVID certificate. Any remaining certificates (if any) chain
+     * the X.509-SVID certificate back to an X.509 root for the trust domain.
+     */
+    List<X509Certificate> chain;
```

Comment:
I think it would be worthwhile to provide a separate property for the leaf certificate. Otherwise users will have to do `getChain().get(0)` in a lot of places, which is less readable than `getLeafCertificate()` or `getLeaf()`.

Reply:
Added method `getLeaf()`